machine/models/baseModel.py

```diff
@@ -0,0 +1,49 @@
+import torch.nn as nn
+import torch.nn.functional as F
+
+class Model(nn.Module):
+    """ 
+    Abstract base class for models.
+
+    Args:
+        encoder_module (baseRNN) :  module that encodes inputs
+        decoder_module (baseRNN, optional):   module to decode encoded inputs
+        decode_function (func, optional): function to generate symbols from output hidden states (default: F.log_softmax)
+
+    """
+
+    def __init__(self, encoder_module, decoder_module=None, decode_function=F.log_softmax):
+        super(Model, self).__init__()
+        self.encoder_module = encoder_module
+        self.decoder_module = decoder_module
+        self.decode_function = decode_function
+
```

Comment:
I would add :

```
@abc.abstractmethod
    def extra_repr(self):
        """Set the extra information about this module for printing.

        Note:
            - Should use `seq2seq.util.herlpers.get_extra_repr`
        """
        pass
```

To force people to use extra_repr, it's a lot cleaner for printing. 

I also defined a cool `get_extra_repr` that automatically makes `extra_repr` for you. Have a look at https://github.com/YannDubs/machine/blob/extrapolator/seq2seq/util/helpers.py#L152 and an example of how i'ts used: https://github.com/YannDubs/machine/blob/extrapolator/seq2seq/attention/mixer.py#L111 and tell us you like it.  Maybe @gautierdag can push it, if not I'll try to push my things this w-e.

Reply:
This would be nice indeed, could you make a feature request for it?